File: common/src/main/java/com/ishow/common/utils/cache/LRUCache.java
package com.ishow.common.utils.cache;

import java.util.LinkedHashMap;

/**
 * Created by yuhaiyang on 2018/9/21.
 * Least Recently Used
 */
public class LRUCache<K, V> extends LinkedHashMap<K, V> {
    private static final long serialVersionUID = 5853563362972200456L;

    private final int MAX_SIZE;

    public LRUCache(int size) {
        super(size, 0.75f, true);
        MAX_SIZE = size;
    }

    @Override
    protected boolean removeEldestEntry(Entry<K, V> eldest) {
        return size() > MAX_SIZE;
    }
}

File: common/src/main/java/com/ishow/common/utils/cache/LRUCache.kt
package com.ishow.common.utils.cache

import java.util.LinkedHashMap

/**
 * Created by yuhaiyang on 2018/9/21.
 * Least Recently Used
 * 最近最久未使用
 */
class LRUCache<K, V>(private val MAX_SIZE: Int) : LinkedHashMap<K, V>(MAX_SIZE, 0.75f, true) {

    override fun removeEldestEntry(eldest: MutableMap.MutableEntry<K, V>): Boolean {
        return size > MAX_SIZE
    }

    companion object {
        private const val serialVersionUID = 5853563362972200456L
    }
}

